payments-core/src/main/java/com/stripe/android/payments/PaymentFlowResultProcessor.kt

```diff
@@ -211,7 +211,7 @@ internal sealed class PaymentFlowResultProcessor<T : StripeIntent, out S : Strip
         var remainingRetries = MAX_RETRIES
 
         var stripeIntentResult = if (shouldCallRefreshIntent(originalIntent)) {
-            refreshStripeIntent(
+            return refreshStripeIntent(
```

Comment:
This looks very confusing: We have a `return` in an assigning `if` statement. Can you tweak this a little?

Reply:
I have a follow up coming that makes this easier to refactor, mind if I do it there?

Reply:
Turns out we were relying on the retry refresh endpoint behavior for wechat pay, so I had to move my follow up into this PR to fix that issue.